src/processTargets/targetUtil/insertionRemovalBehaviors/TokenInsertionRemovalBehavior.ts

```diff
@@ -59,45 +58,81 @@ export function getTokenTrailingDelimiterTarget(
  * removal range is designed to be used with things that should clean themselves
  * up as if they're a range of tokens.
  * @param target The target to get the token removal range for
- * @param contentRange Can be used to override the content range instead of
- * using the one on the target
  * @returns The removal range for the given target
  */
-export function getTokenRemovalRange(
-  target: Target,
-  contentRange?: Range
-): Range {
-  const { document } = target.editor;
-  const actualContentRange = contentRange ?? target.contentRange;
-  const removalRange = getDelimitedSequenceRemovalRange(target, contentRange);
-
-  if (!actualContentRange.isEqual(removalRange)) {
-    const fullRange = expandToFullLine(target.editor, actualContentRange);
-    const fullText = document.getText(fullRange);
-    const fullTextOffset = document.offsetAt(fullRange.start);
-
-    const numTokensContentRangeRemoved = calculateNumberOfTokensAfterRemoval(
-      document,
-      fullText,
-      fullTextOffset,
-      actualContentRange
-    );
-
-    const numTokensRemovalRangeRemoved = calculateNumberOfTokensAfterRemoval(
-      document,
-      fullText,
-      fullTextOffset,
-      removalRange
-    );
-
-    // Using removal range has not merged any tokens. Removal range is ok to use.
-    if (numTokensContentRangeRemoved === numTokensRemovalRangeRemoved) {
-      return removalRange;
+export function getTokenRemovalRange(target: Target): Range {
+  const { editor, contentRange } = target;
+  const { start, end } = contentRange;
+
+  const leadingWhitespaceRange =
+    target.getLeadingDelimiterTarget()?.contentRange ?? makeEmptyRange(start);
+
+  const trailingWhitespaceRange =
+    target.getTrailingDelimiterTarget()?.contentRange ?? makeEmptyRange(end);
+
+  const fullLineRange = expandToFullLine(editor, contentRange);
+
+  if (
+    leadingWhitespaceRange.union(trailingWhitespaceRange).isEqual(fullLineRange)
+  ) {
+    // If we would just be leaving a line with whitespace on it, we delete the
+    // whitespace
+    return fullLineRange;
```

Comment:
This seemed like nice behaviour, so I added it while I was here.  Note that we almost got this behaviour with your version, but not if there was trailing whitespace on the line.  Have now added test for that